lib/alephant/sequencer/sequence_cache.rb

```diff
@@ -0,0 +1,69 @@
+require "dalli-elasticache"
+require "alephant/logger"
+
+module Alephant
+  module Sequencer
+    class SequenceCache
+      include Logger
+
+      attr_reader :config
+
+      DEFAULT_TTL  = 5
+
+      def initialize(config={})
+        @config = config
+
+        unless config_endpoint.nil?
+          @elasticache ||= ::Dalli::ElastiCache.new(config_endpoint, { :expires_in => ttl })
+          @client ||= @elasticache.client
+        else
+          logger.debug "Alephant::SequenceCache::#initialize: No config endpoint, NullClient used"
+          logger.metric "NoConfigEndpoint"
+          @client = NullClient.new
+        end
+      end
+
+      def get(key, &block)
+        begin
+          versioned_key = versioned key
+          result = @client.get versioned_key
+          logger.info "Alephant::SequenceCache#get key: #{versioned_key} - #{result ? 'hit' : 'miss'}"
+          logger.metric "GetKeyMiss" unless result
+          result ? result : set(key, block.call)
+        rescue StandardError => e
+          block.call if block_given?
```

Comment:
We check `if block_given?` here (when we error), and yet we don't check for the blocks existence from within the `begin`'s scope? 

If we were concerned that a block wasn't going to be provided, then I'd expect we'd need the same check when calling `set`